src/gui/plugins/visualize_contacts/VisualizeContacts.cc

```diff
@@ -0,0 +1,376 @@
+  this->dataPtr->positionMarkerMsg.set_visibility(
+    ignition::msgs::Marker::GUI);
+  this->dataPtr->
+    positionMarkerMsg.mutable_lifetime()->
+      set_sec(0);
+  this->dataPtr->
+    positionMarkerMsg.mutable_lifetime()->
+      set_nsec(this->dataPtr->markerLifetime * 1000000);
+
+  this->dataPtr->forceMarkerMsg.set_ns("forces");
+  this->dataPtr->forceMarkerMsg.set_action(ignition::msgs::Marker::ADD_MODIFY);
+  this->dataPtr->forceMarkerMsg.set_type(ignition::msgs::Marker::CYLINDER);
+  this->dataPtr->forceMarkerMsg.set_visibility(ignition::msgs::Marker::GUI);
+  this->dataPtr->forceMarkerMsg.mutable_lifetime()->set_sec(0);
+  this->dataPtr->forceMarkerMsg.mutable_lifetime()->
+    set_nsec(this->dataPtr->markerLifetime * 1000000);
+
+  // Set material properties
+  ignition::msgs::Set(
+    this->dataPtr->positionMarkerMsg.mutable_material()->mutable_ambient(),
+    ignition::math::Color(0, 0, 1, 1));
+  ignition::msgs::Set(
+    this->dataPtr->positionMarkerMsg.mutable_material()->mutable_diffuse(),
+    ignition::math::Color(0, 0, 1, 1));
+
+  ignition::msgs::Set(
+    this->dataPtr->forceMarkerMsg.mutable_material()->mutable_ambient(),
+    ignition::math::Color(0, 1, 0, 1));
+  ignition::msgs::Set(
+    this->dataPtr->forceMarkerMsg.mutable_material()->mutable_diffuse(),
+    ignition::math::Color(0, 1, 0, 1));
+
+  // Set contact position scale
+  ignition::msgs::Set(this->dataPtr->positionMarkerMsg.mutable_scale(),
+    ignition::math::Vector3d(this->dataPtr->contactRadius,
+    this->dataPtr->contactRadius,
+    this->dataPtr->contactRadius));
+}
+
+/////////////////////////////////////////////////
+void VisualizeContacts::OnVisualize(bool _checked)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+  this->dataPtr->checkboxState = _checked;
+}
+
+//////////////////////////////////////////////////
+void VisualizeContacts::Update(const UpdateInfo &_info,
+    EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("VisualizeContacts::Update");
+
+  if (!this->dataPtr->initialized)
+  {
+    // Get the name of the world
+    if (this->dataPtr->worldName.empty())
+    {
+      _ecm.Each<components::World, components::Name>(
+        [&](const Entity &,
+            const components::World *,
+            const components::Name *_name) -> bool
+        {
+          // We assume there's only one world
+          this->dataPtr->worldName = _name->Data();
+          return false;
+        });
+    }
+
+    // Enable collisions
+    this->dataPtr->CreateCollisionData(_ecm);
+    this->dataPtr->initialized = true;
+  }
+
+  {
+    std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+    if (this->dataPtr->checkboxPrevState && !this->dataPtr->checkboxState)
+    {
+      // Remove the markers
+      this->dataPtr->positionMarkerMsg.set_action(
+        ignition::msgs::Marker::DELETE_ALL);
+      this->dataPtr->forceMarkerMsg.set_action(
+        ignition::msgs::Marker::DELETE_ALL);
```

Comment:
When I uncheck the "Show Contacts" box, I get this warning:

`[GUI] [Wrn] [MarkerManager.cc:555] Unable to delete all markers in namespace[forces], namespace can't be found.`

Since we're never creating the forces markers,  I recommend removing all the logic around them. The reason is that it's hard to maintain commented out code. Leaving it out makes the plugin simpler and easier to maintain. Whenever we have support for contact normals and forces and want to show them, we can come back to this PR's history and revive the code. What do you think?